packages/@lwc/engine/src/framework/hot-swaps.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { VM, scheduleRehydration } from './vm';
+import { isFalse, isUndefined, isNull } from '@lwc/shared';
+import { markComponentAsDirty, ComponentConstructor } from './component';
+import { Template } from './template';
+import { StylesheetFactory } from './stylesheet';
+
+const swappedTemplateMap = new WeakMap<Template, Template>();
```

Comment:
Going with Diego's concept of "everything is a module", we could potentially use 1 WeakMap of modules.

Reply:
right, but because the reaction mechanism depends on the type that you're swapping, we have 3 of them.